sphinx/writers/html.py

```diff
@@ -511,10 +497,31 @@ def visit_literal(self, node: Element) -> None:
         if 'kbd' in node['classes']:
             self.body.append(self.starttag(node, 'kbd', '',
                                            CLASS='docutils literal notranslate'))
-        else:
+            return
+        lang = node.get("language", None)
+        if 'code' not in node['classes'] or not lang:
```

Comment:
Should the node only having `language` attribute be represented as an inline code? I think it would be better to judge it only by `code` attribute.

Reply:
The current logic is to highlight only if `code` is in the classes AND the `language` attribute has been set.  If `language` isn't set, we can't highlight since we don't know what language to use.  If `language` is set but `code` is not in classes (not sure how that would happen), then it also won't highlight.

There is the option to use the default language set by the `highlight` directive if the user just uses the plain `code` directive.  That would make the `code` role consistent with the `code-block` directive.  However, that would affect existing uses of the `code` directive, so I think it is probably better not to do that.

Reply:
I undertstand.